processing/src/main/java/org/apache/druid/query/aggregation/SerializablePairLongStringComplexMetricSerde.java

```diff
@@ -141,9 +168,71 @@ public byte[] toBytes(SerializablePairLongString val)
   @Override
   public GenericColumnSerializer<?> getSerializer(SegmentWriteOutMedium segmentWriteOutMedium, String column)
   {
-    return new SerializablePairLongStringColumnSerializer(
-        segmentWriteOutMedium,
-        NativeClearedByteBufferProvider.INSTANCE
-    );
+    if (compressionEnabled) {
+      return new SerializablePairLongStringColumnSerializer(
+          segmentWriteOutMedium,
+          NativeClearedByteBufferProvider.INSTANCE
+      );
+    } else {
+      return LargeColumnSupportedComplexColumnSerializer.create(
+          segmentWriteOutMedium, column, LEGACY_STRATEGY
+      );
+    }
   }
+
+  private static final ObjectStrategy<SerializablePairLongString> LEGACY_STRATEGY =
```

Comment:
nit: maybe worth leaving a comment about why this is different from `getObjectStrategy()` here too (sad all big endian all the time, etc)